File: week1/reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        // Create pointers at each end of string
        int hi = s.length - 1;
        int lo = 0;

        // Move pointers toward center at an equal rate
        while (hi > lo) {
            // Switch characters
            char temp = s[lo];
            s[lo] = s[hi];
            s[hi] = temp;
            hi--;
            lo++;
        }
    }
}

File: spring2020/unit1/reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        lo = 0
        hi = len(s) - 1
        while lo < hi:
            temp = s[lo]
            s[lo] = s[hi]
            s[hi] = temp
            lo += 1
            hi -= 1
        